modules/tensor_mechanics/include/utils/RankFourTensor.h

```diff
@@ -305,4 +312,27 @@ class RankFourTensor
 
 inline RankFourTensor operator*(Real a, const RankFourTensor & b) { return b * a; }
 
+template<class T>
+void
+RankFourTensor::rotate(T & R)
+{
+  RankFourTensor old = *this;
+
+  for (unsigned int i = 0; i < N; ++i)
+    for (unsigned int j = 0; j < N; ++j)
+      for (unsigned int k = 0; k < N; ++k)
+        for (unsigned int l = 0; l < N; ++l)
+        {
+          Real sum = 0.0;
+          for (unsigned int m = 0; m < N; ++m)
+            for (unsigned int n = 0; n < N; ++n)
+              for (unsigned int o = 0; o < N; ++o)
+                for (unsigned int p = 0; p < N; ++p)
+                  sum += R(i,m) * R(j,n) * R(k,o) * R(l,p) * old(m,n,o,p);
+
+          _vals[i][j][k][l] = sum;
+        }
+}
+
+
```

Comment:
Even though at the group meeting Wen will probably say I helped only with "style issues" (despite having added a whole new class of user objects fixing threading issues) I will point out the extra empty line here.